fuel-relayer/src/validators.rs

```diff
@@ -0,0 +1,128 @@
+use fuel_core_interfaces::{
+    model::{DaBlockHeight, ValidatorStake},
+    relayer::RelayerDb,
+};
+use fuel_tx::Address;
+use std::collections::{hash_map::Entry, HashMap};
+use tracing::{error, info};
+
+/// It contains list of Validators and its stake and consensus public key.
+/// We dont expect big number of validators in that sense we are okey to have it all in memory
+/// for fast access. Data availability height (da_height) represent snapshot height that set represent.
+/// Validator Set is same as it is inside database.
+#[derive(Default)]
+pub struct Validators {
+    /// Validator set
+    pub set: HashMap<Address, (ValidatorStake, Option<Address>)>,
+    /// Da height
+    pub da_height: DaBlockHeight,
+}
+
+impl Validators {
+    // probably not going to metter a lot we expect for validator stake to be mostly unchanged.
+    // TODO if it takes a lot of time to load it is good to optimize.
+    pub async fn load(&mut self, db: &dyn RelayerDb) {
+        self.da_height = db.get_validators_da_height().await;
+        self.set = db.get_validators().await;
+    }
+
+    /// Get validator set
+    pub async fn get(
+        &mut self,
+        da_height: DaBlockHeight,
+    ) -> Option<HashMap<Address, (u64, Option<Address>)>> {
+        // TODO apply down drift https://github.com/FuelLabs/fuel-core/issues/365
+        if self.da_height == da_height {
+            return Some(self.set.clone());
```

Comment:
is cloning the entire set the best option here?

Reply:
Yeah, this function is called in request by `fuel-bft` and the relayer need to return the copy of the requested validator set to it. 
I will implement drift in past in this commit: https://github.com/FuelLabs/fuel-core/issues/365